Foundation/Decimal.swift

```diff
@@ -661,17 +661,19 @@ extension Decimal {
     public mutating func formTruncatingRemainder(dividingBy other: Decimal) { fatalError("Decimal does not yet fully adopt FloatingPoint") }
 }
 
-extension Decimal: CustomStringConvertible {
+extension Decimal : CustomStringConvertible {
     public init?(string: String, locale: Locale? = nil) {
         let scan = Scanner(string: string)
-        scan.locale = locale
         var theDecimal = Decimal()
+        scan.locale = locale
         if !scan.scanDecimal(&theDecimal) {
             return nil
         }
         self = theDecimal
     }
 
+    // Note: In the Darwin overlay, `description` is implemented in terms of
+    // `NSDecimalString(_:_:)`; here, it's the other way around.
```

Comment:
We should do the same here and implement the conversion in `NSDecimalString(_:_:)` since it takes a `Locale`. Then `NSDecimalNumber.description(withLocale:)`, `NSDecimalNumber.description` and `Decimal.description` can all call `NSDecimalString(_:_:)` to match the Darwin overlay and have localisation work correctly as well. Do you want to split out a PR to do this change? You can ignore the localisation part for now as it can be done in a followup to keep it as NFC.

Reply:
Yes, that would be an optimal arrangement.

I have some additional gardening planned for the portions specific to the sclf version of this file. After that, I think the way to go would be some PRs to implement `NSDecimalString` and other `NSDecimal*` functions without using overlay methods, then to switch over the overlay implementations to call the `NSDecimal*` methods. There are several other divergences (`compare(to:)` versus `NSDecimalCompare`) that could benefit from this approach.

Reply:
FYI Ive done an implementation of `NSDecimalString` that also takes the `locale` into account so once this PR is merged I will put it up.

Reply:
That's great news!